Update assemblies and modules Docs for Metrics Reporter integration (#11329)
strimzi/strimzi-kafka-operator: documentation/assemblies/metrics/assembly-metrics-config-files.adoc

│    ├── prometheus-rules.yaml <7>
│    ├── prometheus.yaml <8>
│    └── strimzi-pod-monitor.yaml <9>
├── strimzi-metrics-reporter
|    ├── grafana-dashboards <10>
|    |  ├── strimzi-kafka.json
|    |  ├── strimzi-kraft.json
|    └── kafka-metrics.yaml <11>
├── kafka-bridge-metrics.yaml <12>
├── kafka-connect-metrics.yaml <13>
├── kafka-cruise-control-metrics.yaml <14>
├── kafka-metrics.yaml <15>
├── kafka-mirror-maker-2-metrics.yaml <16>

> └── oauth-metrics.yaml <17>

Reviewer: a couple of questions here:

* aren't we getting examples for all the other components (apart Kafka) for using the Strimzi Metrics reporter?
* wouldn't be better to put the Prometheus JMX exporter related example in a dedicated folder now that we have the `strimzi-metrics-reporter` one?

Reviewer: Yeah that was the idea thanks @OwenCorrigan76 
@fvaleri @PaulRMellor wdyt?

Reviewer 2: LGTM, but the folder name should be `prometheus-jmx-exporter` and we need a similar README.md for consistency.

Author: Yes. I'll update this docs PR and then update the main integration PR. Thanks

--

> <1> Example Grafana dashboards for the different Strimzi components.

Reviewer 3: ```suggestion
<1> Grafana dashboards for components using the JMX Exporter.
```

<7> Alerting rules examples for use with Prometheus Alertmanager (deployed with Prometheus).
<8> Installation resource file for the Prometheus image.
<9> PodMonitor definitions translated by the Prometheus Operator into jobs for the Prometheus server to be able to scrape metrics data directly from pods.

> <10> Example Grafana dashboards with Metrics Reporter for the Kafka component.

Reviewer 2: ```suggestion
<10> Example Grafana dashboards to be used with Metrics Reporter for the Kafka component.
```

> <11> Resource for deploying and configuring Strimzi Metrics Reporter based Kafka.

Reviewer 2: ```suggestion
<11> Resource for deploying and configuring Kafka with Strimzi Metrics Reporter.
```

<7> Alerting rules examples for use with Prometheus Alertmanager (deployed with Prometheus).
<8> Installation resource file for the Prometheus image.
<9> PodMonitor definitions translated by the Prometheus Operator into jobs for the Prometheus server to be able to scrape metrics data directly from pods.

> <10> Grafana dashboards for components using the Metrics Reporter.

Reviewer 4: ```suggestion
<10> Grafana dashboards for components using the Strimzi Metrics Reporter.
```

> <10> Example Grafana dashboards with Metrics Reporter for the Kafka component.

Reviewer 3: ```suggestion
<10> Grafana dashboards for components using the Metrics Reporter.
```

<11> Resource for deploying and configuring Strimzi Metrics Reporter based Kafka.
<12> Kafka Bridge resource with metrics enabled.
<13> Metrics configuration that defines Prometheus JMX Exporter relabeling rules for Kafka Connect.
<14> Metrics configuration that defines Prometheus JMX Exporter relabeling rules for Cruise Control.
<15> Metrics configuration that defines Prometheus JMX Exporter relabeling rules for Kafka.
<16> Metrics configuration that defines Prometheus JMX Exporter relabeling rules for MirrorMaker 2.
<17> Metrics configuration that defines Prometheus JMX Exporter relabeling rules for OAuth 2.0.

NOTE: The Prometheus JMX Exporter collects Java Management Extensions (JMX) from Kafka components and converts them into Prometheus metrics.
You do not require `jmxOptions` configuration in the custom resource of the component for the Prometheus JMX Exporter to function.
`jmxOptions` is only required if you need direct access to JMX from Kafka components.

> Strimzi uses the StrimziMetricsReporter or the {JMXExporter} to expose metrics through an HTTP endpoint, which can be scraped by the Prometheus server.

Reviewer 5: Should Strimzi Metrics Reporter also have a link?
```suggestion
Strimzi uses the {StrimziMetricsReporter} or the {JMXExporter} to expose metrics through an HTTP endpoint, which can be scraped by the Prometheus server.
```
